syntax = "proto3";

service Booking{

  // Fonction crée par Tournier Quentin et Marche Jules
  // But: Afficher toutes les réservations d'un Utilisateur
  // En entrée: UserId
  // En sortie: Un objet regroupant toutes les réservation de l'utilisateur
  rpc GetBookingByUserid(BookingUserId) returns (BookingObject) {}


  // Fonction crée par Tournier Quentin et Marche Jules
  // But: Ajouter une réservation à un utilisateur
  // En entrée: L'objet OneBooking:
  //                -userId
  //                -date
  //                -movieId
  // En sortie: Un message stipulant l'ajout de la réservation ou d'erreur
  rpc AddBookingByUserid(OneBooking) returns (NotificationMessageBooking) {}
}

message BookingObject{
  string userid = 1;
  repeated Dates dates = 2;
}

message Dates{
  string date = 1;
  repeated string movieid = 2;
}

message OneBooking{
  string userid = 1;
  string date = 2;
  string movieid = 3;
}

message NotificationMessageBooking{
  string message = 1;
}

message BookingUserId{
  string userid = 1;
}